### ISSUE TITLE
Web on chain broken

### ISSUE THREAD

**Issue by Thykof** [massalabs/station#1099]
**The problem(s)**

I tried to upload a website put the API call failed and I see in the logs:

```
2023/08/14 16:41:04 http2: panic serving 127.0.0.1:50280: fetching the 'index.html' web resource at 'AS12YDfkUdbCksHKCaMtdCBbMscrUEtf53hE772eugb2LaSdBdGaL': fetching website content at AS12YDfkUdbCksHKCaMtdCBbMscrUEtf53hE772eugb2LaSdBdGaL from blockchain: invalid chunk size for chunk 0
goroutine 1316 [running]:
net/http.(*http2serverConn).runHandler.func1()
	/opt/homebrew/Cellar/go/1.20.6/libexec/src/net/http/h2_bundle.go:6042 +0x138
panic({0x1022204c0, 0x140002101c0})
	/opt/homebrew/Cellar/go/1.20.6/libexec/src/runtime/panic.go:884 +0x204
github.com/massalabs/station/pkg/onchain/website.Request({0x10230dee0, 0x140000c6798}, 0x1400003e4b0?, 0x21?, {0x14000558300?, 0x35?}, {0x10100d9d8, 0xa})
	/Users/nathanmassa/dev/station/pkg/onchain/website/browser.go:28 +0xc0
github.com/massalabs/station/pkg/onchain/website.handleMassaDomainRequest({0x10230dee0, 0x140000c6798}, 0x1400016f600, 0x3, {{0x14000110ec8, 0x8}, {0x1400003e4b0, 0x21}, {0x14000040680, 0x35}})
	/Users/nathanmassa/dev/station/pkg/onchain/website/handler.go:30 +0x120
github.com/massalabs/station/pkg/onchain/website.MassaTLDInterceptor(0x140017cfc10, {{0x14000110ec8, 0x8}, {0x1400003e4b0, 0x21}, {0x14000040680, 0x35}})
	/Users/nathanmassa/dev/station/pkg/onchain/website/handler.go:42 +0xd8
github.com/massalabs/station/api.TopMiddleware.func1({0x10230dee0, 0x140000c6798}, 0x1400016f600)
	/Users/nathanmassa/dev/station/api/middleware.go:25 +0x110
net/http.HandlerFunc.ServeHTTP(0x0?, {0x10230dee0?, 0x140000c6798?}, 0x100a398c8?)
	/opt/homebrew/Cellar/go/1.20.6/libexec/src/net/http/server.go:2122 +0x38
net/http.serverHandler.ServeHTTP({0x1400058bd78?}, {0x10230dee0, 0x140000c6798}, 0x1400016f600)
	/opt/homebrew/Cellar/go/1.20.6/libexec/src/net/http/server.go:2936 +0x2d8
net/http.initALPNRequest.ServeHTTP({{0x10230e7a0?, 0x140007cb2c0?}, 0x14000329180?, {0x140001fc0f0?}}, {0x10230dee0, 0x140000c6798}, 0x1400016f600)
	/opt/homebrew/Cellar/go/1.20.6/libexec/src/net/http/server.go:3545 +0x1f4
net/http.(*http2serverConn).runHandler(0x102cd7c08?, 0x0?, 0x0?, 0x0?)
	/opt/homebrew/Cellar/go/1.20.6/libexec/src/net/http/h2_bundle.go:6049 +0x78
created by net/http.(*http2serverConn).processHeaders
	/opt/homebrew/Cellar/go/1.20.6/libexec/src/net/http/h2_bundle.go:5762 +0x5c4
```

when I click on a website to visit it:
```
2023/08/14 17:08:38 http: panic serving 127.0.0.1:50658: fetching the 'index.html' web resource at 'AS12fCEQPW5kLMWABrRSgsGDnLQZkfji7WZAEU9R8wR7TsGp1srHF': fetching website content at AS12fCEQPW5kLMWABrRSgsGDnLQZkfji7WZAEU9R8wR7TsGp1srHF from blockchain: invalid chunk size for chunk 0
goroutine 1106 [running]:
net/http.(*conn).serve.func1()
	/opt/homebrew/Cellar/go/1.20.6/libexec/src/net/http/server.go:1854 +0xb0
panic({0x1022284c0, 0x140004281e0})
	/opt/homebrew/Cellar/go/1.20.6/libexec/src/runtime/panic.go:890 +0x258
github.com/massalabs/station/pkg/onchain/website.Request({0x102315fa0, 0x140005c0b60}, 0x140000dc4b0?, 0x21?, {0x140008d8600?, 0x35?}, {0x1010159d8, 0xa})
	/Users/nathanmassa/dev/station/pkg/onchain/website/browser.go:28 +0xc0
github.com/massalabs/station/pkg/onchain/website.handleMassaDomainRequest({0x102315fa0, 0x140005c0b60}, 0x14000a19d00, 0xd, {{0x14000210df8, 0x8}, {0x140000dc4b0, 0x21}, {0x14000040680, 0x35}})
	/Users/nathanmassa/dev/station/pkg/onchain/website/handler.go:30 +0x120
github.com/massalabs/station/pkg/onchain/website.MassaTLDInterceptor(0x1400161fa40, {{0x14000210df8, 0x8}, {0x140000dc4b0, 0x21}, {0x14000040680, 0x35}})
	/Users/nathanmassa/dev/station/pkg/onchain/website/handler.go:42 +0xd8
github.com/massalabs/station/api.TopMiddleware.func1({0x102315fa0, 0x140005c0b60}, 0x14000a19d00)
	/Users/nathanmassa/dev/station/api/middleware.go:25 +0x110
net/http.HandlerFunc.ServeHTTP(0x0?, {0x102315fa0?, 0x140005c0b60?}, 0x100ae1c48?)
	/opt/homebrew/Cellar/go/1.20.6/libexec/src/net/http/server.go:2122 +0x38
net/http.serverHandler.ServeHTTP({0x1400044fec0?}, {0x102315fa0, 0x140005c0b60}, 0x14000a19d00)
	/opt/homebrew/Cellar/go/1.20.6/libexec/src/net/http/server.go:2936 +0x2d8
net/http.(*conn).serve(0x140004f9ef0, {0x1023167a0, 0x140006600f0})
	/opt/homebrew/Cellar/go/1.20.6/libexec/src/net/http/server.go:1995 +0x560
created by net/http.(*Server).Serve
	/opt/homebrew/Cellar/go/1.20.6/libexec/src/net/http/server.go:3089 +0x520
```

**To Reproduce**

**Expected behavior(s)**

**Screenshots**

**MassaStation version**
The version of MassaStation the bug is present in.

main branch

**Desktop (please complete the following information):**
 - OS: [e.g. MacOS, Windows...] any
 
 **Any other useful info**

## QA
- upload a website
- navigate into it.

**Comment by Thykof** [comment]
The pull request merged (#1100) avoid panic but doesn't solve the issue.

Now, I can upload a website but when I try to navigate in it:

```
2023-08-21T15:56:54.981+0200	ERROR	logger/global.go:106	handling massa domain request: fetching the 'index.html' web resource at 'AS1N9L9KzLQikrxeq9Hyo9QuMpvrwBfKGQccNTRYwvVhCUq2ZwXU': fetching the 'index.html' web resource at 'AS1N9L9KzLQikrxeq9Hyo9QuMpvrwBfKGQccNTRYwvVhCUq2ZwXU': fetching website content at AS1N9L9KzLQikrxeq9Hyo9QuMpvrwBfKGQccNTRYwvVhCUq2ZwXU from blockchain: invalid chunk size for chunk 0
```

**Comment by Thykof** [comment]
> Previously uploaded websites still works, so the bug is in the uploading process.

No, the problem comes from the fact that it can take a lot of time for the content of the website.

We can probably improve this by reading final value from the smart contract datastore instead of candidate value
